lib/main.dart

```diff
@@ -30,32 +31,87 @@ class CircuitVerseMobile extends StatelessWidget {
+                  fontFamily: 'Poppins',
+                  cursorColor: CVTheme.primaryColor,
+                ),
+                description: 'LightTheme',
+              ),
+              AppTheme(
+                id: 'dark',
+                data: ThemeData(
+                  primaryColor: CVTheme.secondaryColor,
+                  accentColor: CVTheme.secondaryColor,
+                  fontFamily: 'Poppins',
+                  brightness: Brightness.dark,
+                  cursorColor: CVTheme.primaryColor,
+                ),
+                description: 'DarkTheme',
+              ),
+            ],
+            child: ThemeConsumer(
+              child: Builder(
+                builder: (themeContext) => GetMaterialApp(
+                  theme: ThemeProvider.themeOf(themeContext).data,
+                  title: 'CircuitVerse Mobile',
+                  localizationsDelegates: [
+                    AppLocalizationsDelegate(),
+                    GlobalMaterialLocalizations.delegate,
+                    GlobalWidgetsLocalizations.delegate,
+                  ],
+                  supportedLocales: [
+                    Locale('en', ''),
+                  ],
+                  onGenerateTitle: (BuildContext context) =>
+                      AppLocalizations.of(context).title,
+                  debugShowCheckedModeBanner: false,
+                  builder: (context, child) => Navigator(
+                    key: locator<DialogService>().dialogNavigationKey,
+                    onGenerateRoute: (settings) => MaterialPageRoute(
+                      builder: (context) => DialogManager(child: child),
+                    ),
+                  ),
+                  onGenerateRoute: CVRouter.generateRoute,
+                  home: StartUpView(),
+                ),
+              ),
+            ),
+          )
+        : GetMaterialApp(
+            title: 'CircuitVerse Mobile',
+            localizationsDelegates: [
+              AppLocalizationsDelegate(),
+              GlobalMaterialLocalizations.delegate,
+              GlobalWidgetsLocalizations.delegate,
+            ],
+            supportedLocales: [
+              Locale('en', ''),
+            ],
+            onGenerateTitle: (BuildContext context) =>
+                AppLocalizations.of(context).title,
+            debugShowCheckedModeBanner: false,
+            builder: (context, child) => Navigator(
+              key: locator<DialogService>().dialogNavigationKey,
+              onGenerateRoute: (settings) => MaterialPageRoute(
+                builder: (context) => DialogManager(child: child),
+              ),
+            ),
+            onGenerateRoute: CVRouter.generateRoute,
+            theme: ThemeData(
+              primarySwatch: generateMaterialColor(CVTheme.primaryColor),
+              fontFamily: 'Poppins',
+              cursorColor: CVTheme.primaryColor,
+            ),
+            home: StartUpView(),
+          );
   }
 }
```

Comment:
Code repetition for `GetMaterialApp` makes it look messy. Can be refactored.

Reply:
i added so that ui test doesn't fail

Reply:
@iCoder-007 If you don't use condition then UI Tests fails ?

Reply:
yes that is why i added it

Reply:
I just now removed this condition and using `ThemeProvider` code, the UI Tests are passing.

<details>
<summary> Here is the diff </summary>

```
diff --git a/lib/main.dart b/lib/main.dart
index 3c7c4cb..c6161d9 100644
--- a/lib/main.dart
+++ b/lib/main.dart
@@ -31,8 +31,7 @@ class CircuitVerseMobile extends StatelessWidget {
       DeviceOrientation.portraitDown,
     ]);
 
-    return ThemeProvider != null
-        ? ThemeProvider(
+    return ThemeProvider(
             saveThemesOnChange: true,
             loadThemeOnInit: true,
             themes: [
@@ -85,33 +84,6 @@ class CircuitVerseMobile extends StatelessWidget {
                 ),
               ),
             ),
-          )
-        : GetMaterialApp(
-            title: 'CircuitVerse Mobile',
-            localizationsDelegates: [
-              AppLocalizationsDelegate(),
-              GlobalMaterialLocalizations.delegate,
-              GlobalWidgetsLocalizations.delegate,
-            ],
-            supportedLocales: [
-              Locale('en', ''),
-            ],
-            onGenerateTitle: (BuildContext context) =>
-                AppLocalizations.of(context).title,
-            debugShowCheckedModeBanner: false,
-            builder: (context, child) => Navigator(
-              key: locator<DialogService>().dialogNavigationKey,
-              onGenerateRoute: (settings) => MaterialPageRoute(
-                builder: (context) => DialogManager(child: child),
-              ),
-            ),
-            onGenerateRoute: CVRouter.generateRoute,
-            theme: ThemeData(
-              primarySwatch: generateMaterialColor(CVTheme.primaryColor),
-              fontFamily: 'Poppins',
-              cursorColor: CVTheme.primaryColor,
-            ),
-            home: StartUpView(),
           );
   }
 }

```

</details>

Reply:
so should i remove the condition

Reply:
Yes

Reply:
Also test the changes on your device once.

Reply:
I just tried the ui test after removing condition it shows all test passes

Reply:
Looks far better now. Nice.